File: Searcha2DMatrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int left = 0;
        int right = matrix[0].length-1;
        int matrixlen = 0;
        while(matrixlen < matrix.length){
            if(matrix[matrixlen][left] == target){
                return true;
            }
            if(matrix[matrixlen][right] == target){
                return true;
            }
            left++;
            right--;
            if(left > right){
                matrixlen++;
                left = 0;
                right = matrix[0].length-1;
            }
        }
        return false;
    }
}

File: Searcha2DMatrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        row = 0
        left = 0
        right = len(matrix[0])-1
        while(row < len(matrix)):
            mid = (left+right)/2
            if(matrix[row][mid] == target):
                return True
            elif(matrix[row][mid] < target):
                left = mid + 1
            else:
                right = mid - 1
            if(left > right):
                left = 0
                right = len(matrix[0])-1
                row += 1
        return False
        